src/main/java/seedu/guest/model/guest/UniqueGuestList.java

```diff
@@ -63,6 +74,9 @@ public void setGuest(Guest target, Guest editedGuest) {
 
         if (!target.isSameGuest(editedGuest) && contains(editedGuest)) {
             throw new DuplicateGuestException();
+            // Only check for room equality if the guest's room has been edited
+        } else if (!target.getRoom().equals(editedGuest.getRoom()) && hasSameRoom(editedGuest)) {
```

Comment:
Can we place the comment in the `else if` block?

Reply:
Agreed!